odbc/src/api/data.rs

```diff
@@ -1317,4 +1368,209 @@ mod unit {
             );
         }
     }
+
+    // checks we properly parse time data
+    mod convert_to_time {
```

Comment:
Maybe you can add a test case for each toXX, where the syntax is correct, but the data is not valid.
Something like `25:10:32` for example.
This is covered by the code, but not tested. I think it's good to show that we make sure that the data is also valid on top of checking the format.

Reply:
good idea, will do!